From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nihal Jere <njere@uprev.com>
Date: Tue, 2 Apr 2024 13:31:51 -0500
Subject: [PATCH] 6376: ArrayDataType: add constructor without elementLength
 parameter

The elementLength parameter is unused when the base type isn't dynamic, but the existing constructors require it to be specified.
---
 .../java/ghidra/program/model/data/ArrayDataType.java    | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/model/data/ArrayDataType.java b/Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/model/data/ArrayDataType.java
index b280085b17..31fb8af3f7 100644
--- a/Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/model/data/ArrayDataType.java
+++ b/Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/model/data/ArrayDataType.java
@@ -32,6 +32,15 @@ public class ArrayDataType extends DataTypeImpl implements Array {
 	private int elementLength;
 	private boolean deleted = false;
 
+	/**
+	 * Constructs a new Array dataType.
+	 * @param dataType the dataType of the elements in the array (null is not permitted).
+	 * @param numElements the number of elements in the array (0 is permitted).
+	 */
+	public ArrayDataType(DataType dataType, int numElements) {
+		this(dataType, numElements, -1, null);
+	}
+
 	/**
 	 * Constructs a new Array dataType.
 	 * @param dataType the dataType of the elements in the array (null is not permitted).
-- 
2.44.0

